frontend/packages/metal3-plugin/src/components/baremetal-hosts/BareMetalHostStatus.tsx

```diff
@@ -48,10 +53,35 @@ const BareMetalHostStatus: React.FC<StatusProps> = ({ status, title, ...props })
       return <AddDiscoveredHostButton host={props.host} />;
     case [NODE_STATUS_STARTING_MAINTENANCE, NODE_STATUS_UNDER_MAINTENANCE].includes(status):
       return <MaintenancePopover title={statusTitle} maintenance={props.maintenance} />;
+    case status === HOST_STATUS_INSPECTING:
+      return (
+        <ProgressStatus title={statusTitle}>
+          The hardware details for the host are being collected by an agent. This operation will
+          take a while. The host will become available when finished.
+        </ProgressStatus>
+      );
+    case status === HOST_STATUS_PROVISIONING:
+      return (
+        <ProgressStatus title={statusTitle}>
+          An image is being written to the host&apos;s disk(s).
+        </ProgressStatus>
+      );
+    case status === HOST_STATUS_DEPROVISIONING:
+      return (
+        <ProgressStatus title={statusTitle}>
+          The image is being wiped from the host&apos;s disk(s).
+        </ProgressStatus>
+      );
     case [NODE_STATUS_STOPPING_MAINTENANCE, ...HOST_PROGRESS_STATES].includes(status):
       return <ProgressStatus title={statusTitle} />;
     case HOST_ERROR_STATES.includes(status):
       return <ErrorStatus title={statusTitle}>{getHostErrorMessage(props.host)}</ErrorStatus>;
+    case [HOST_STATUS_AVAILABLE, HOST_STATUS_READY].includes(status):
+      return (
+        <SuccessStatus title={statusTitle}>
+          The host is available to be provisioned as a node.
```

Comment:
```suggestion
          The host is available to be provisioned as a node. If the ClusterAutoscaler is configured, it will automatically be used as a worker node when the cluster needs additional resources.
```

I wish we could add a `Learn more about autoscaling` link here too, but I'm not sure how to reliably link to the latest documentation for it. [4.2's documentation are here](https://docs.openshift.com/container-platform/4.2/machine_management/applying-autoscaling.html), but that might be outdated for 4.3 where this message would be seen.

It'd be great if we could go "up the chain" from BareMetalHost->Machine->MachineSet->MachineAutoscaler->ClusterAutoscaler to see if this host is properly configured to be provisioned by the cluster when it's needed, and then show a unique message here with a link to the relevant MachineAutoscaler resource.

I'm not sure if that's doable though, and it could probably be a future enhancement anyway.

Reply:
I don't think there's any support for bare metal in the ClusterAutoscaler, nor does it likely make sense in the foreseeable future. Even if it identified that the cluster needs more workers, we're looking at 30+ minutes before you could reasonably expect a host to be provisioned and ready for work, maybe more depending on the hardware. The latency and expense involved in adding and removing bare metal hosts just isn't worth the effort.

In the future we may be able to support some kind of hibernate or suspend behavior for bare metal hosts when not needed, but that's merely a twinkle in a PM's eye right now.

Reply:
> but that's merely a twinkle in a PM's eye right now.

😄 

Thank you @mhrivnak, that makes total sense to me. I keep forgetting we're not in the future quite yet... although it _is_ 2020.

I suppose this also means the MachineAutoscalers we include in the navigation of the Web Console are largely useless for bare metal right now. Interesting. I suspect we won't bother hiding them in the nav for the moment, but we definitely shouldn't mention autoscaling on bare metal in this descriptive text right now. Ignore my text suggestion @jtomasek. 👍